syntax = "proto3";

package blockstorage;

service BlockStorage {
  rpc Read (ReadRequest) returns (ReadReply) {}
  rpc Write (WriteRequest) returns (WriteReply) {}
}

message ReadRequest {
  uint32 addr = 1;
}

message ReadReply {
  string buffer = 1;
  uint32 error = 2;
}

message WriteRequest {
  uint32 addr = 1;
  string buffer = 2;
}

message WriteReply {
  uint32 error = 1;
}